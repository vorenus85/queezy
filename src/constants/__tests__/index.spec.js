import { describe, it, expect, test } from 'vitest'
import { pages, difficulty, difficultyLength, categories } from '../index.js'

describe('It should be correct page names', () => {
  test.each`
    name                | value
    ${'start'}          | ${'START'}
    ${'difficulty'}     | ${'DIFFICULTY'}
    ${'questions'}      | ${'QUESTIONS'}
    ${'result'}         | ${'RESULT'}
    ${'detailedResult'} | ${'DETAILED_RESULTS'}
  `('page: $name, should be: $value', ({ name, value }) => {
    const actualResult = pages[name]

    expect(actualResult).toBe(value)
  })
})

describe('It should be correct difficulty names', () => {
  test.each`
    name        | value
    ${'easy'}   | ${'EASY'}
    ${'medium'} | ${'MEDIUM'}
    ${'hard'}   | ${'HARD'}
  `('difficulty: $name, should be: $value', ({ name, value }) => {
    const actualResult = difficulty[name]

    expect(actualResult).toBe(value)
  })
})

describe('It should be correct difficulty lengths', () => {
  test.each`
    name        | value
    ${'easy'}   | ${15}
    ${'medium'} | ${10}
    ${'hard'}   | ${6}
  `('difficultyLength: $name, should be: $value', ({ name, value }) => {
    const actualResult = difficultyLength[name]

    expect(actualResult).toBe(value)
  })
})

describe('Categories tests', () => {
  it('should be 6 quiz categories', () => {
    expect(categories.length).toBe(6)
  })

  it('quiz categories matches the array snapshot', () => {
    const quizCategories = categories

    expect(quizCategories).toMatchSnapshot()
  })
})
