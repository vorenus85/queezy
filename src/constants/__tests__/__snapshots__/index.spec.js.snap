// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Array snapshot tests > matches the array snapshot 1`] = `
[
  {
    "key": "film/series",
    "title": "Film / Series",
  },
  {
    "key": "gastronomy",
    "title": "Gastronomy",
  },
  {
    "key": "history",
    "title": "History",
  },
  {
    "key": "science",
    "title": "Science",
  },
  {
    "key": "sport",
    "title": "Sport",
  },
  {
    "key": "mixed",
    "title": "Mixed",
  },
]
`;

exports[`Categories snapshot tests > matches the array snapshot 1`] = `
[
  {
    "key": "film/series",
    "title": "Film / Series",
  },
  {
    "key": "gastronomy",
    "title": "Gastronomy",
  },
  {
    "key": "history",
    "title": "History",
  },
  {
    "key": "science",
    "title": "Science",
  },
  {
    "key": "Sport",
    "title": "Sport",
  },
  {
    "key": "mixed",
    "title": "Mixed",
  },
]
`;

exports[`Categories tests > quiz categories matches the array snapshot 1`] = `
[
  {
    "key": "film/series",
    "title": "Film / Series",
  },
  {
    "key": "gastronomy",
    "title": "Gastronomy",
  },
  {
    "key": "history",
    "title": "History",
  },
  {
    "key": "science",
    "title": "Science",
  },
  {
    "key": "Sport",
    "title": "Sport",
  },
  {
    "key": "mixed",
    "title": "Mixed",
  },
]
`;
